src/mqt/bench/benchmark_generator.py

```diff
@@ -249,17 +249,59 @@ def generate_indep_levels(
                     break
 
 
+@overload
+def get_benchmark(
+    benchmark_name: str,
+    level: str | int,
+    circuit_size: int | None = None,
+    benchmark_instance_name: str | None = None,
+    compiler: Literal["qiskit"] = "qiskit",
+    compiler_settings: CompilerSettings | None = None,
+    gate_set_name: str = "ibm",
+    device_name: str = "ibm_washington",
+) -> QuantumCircuit:
+    ...
+
+
+@overload
+def get_benchmark(
+    benchmark_name: str,
+    level: str | int,
+    circuit_size: int | None = None,
+    benchmark_instance_name: str | None = None,
+    compiler: Literal["tket"] = "tket",
+    compiler_settings: CompilerSettings | None = None,
+    gate_set_name: str = "ibm",
+    device_name: str = "ibm_washington",
+) -> Circuit:
+    ...
+
+
+@overload
+def get_benchmark(
+    benchmark_name: str,
+    level: str | int,
+    circuit_size: int | None = None,
+    benchmark_instance_name: str | None = None,
+    compiler: str = "qiskit",
+    compiler_settings: CompilerSettings | None = None,
+    gate_set_name: str = "ibm",
+    device_name: str = "ibm_washington",
+) -> QuantumCircuit | Circuit:
+    ...
```

Comment:
Maybe I am stupid, but that overload should not be necessary as the function definition below is identical.
You can just remove it entirely.

```suggestion
```

Reply:
Removing that leads to the following error: 
`error: No overload variant of "get_benchmark" matches
argument types "str", "Union[str, int]", "int", "Optional[str]", "str",
"Optional[CompilerSettings]", "str", "str"  [call-overload]
`
Do you know how to otherwise avoid that?

Reply:
Hm. strange. Is there some difference between the overload and the actual function signature down below.
I remember such duplicate overloads not being necessary in our other tools (like QMAP).

Reply:
Nope, it is the exact same. 

I am just a bit confused by the error message itself. The third parameter (`circuit_size: int | None = None,)` is not described as `Optional[int]` but only `int` while for the other optional ones it is different. Do you have an idea why?

Reply:
To be honest: not really... maybe because it is called with an `int`?

Reply:
I re-added the overload variant that copies the function since apparently it is best practice and also done here: https://mypy.readthedocs.io/en/stable/literal_types.html

Reply:
Alright 👌🏻